cylc/uiserver/authorise.py

```diff
@@ -0,0 +1,476 @@
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from contextlib import suppress
+from functools import lru_cache
+import grp
+from typing import List, Dict, Optional, Union, Any, Sequence, Set
+from inspect import iscoroutinefunction
+import logging
+import os
+from tornado import web
+from traitlets.config.loader import LazyConfigValue
+from traitlets.traitlets import Bool
+
+logger = logging.getLogger(__name__)
+
+
+class Authorization:
+    """Authorization Information Class
+    One instance for the life of the UI Server. If authorization settings
+    change they will need to re-start the UI Server.
+    Authorization has access groups: `READ`, `CONTROL`, `ALL` - along with
+    their negations, `!READ`, `!CONTROL` and `!ALL` which indicate removal of
+    the permission groups.
+
+    """
+
+    # config literals
+    DEFAULT = 'default'
+    LIMIT = 'limit'
+    GRP_IDENTIFIER = 'group:'
```

Comment:
Wouldn't it be useful to have identifier for user and group? I'm using the sample configuration you provided, and searching for "group:group1" is quite useful. But if I have a user with name "group1", then searching will find both.

Reply:
I can see adding `user:` would be very clear. For info, this was discussed and was ruled out as it is additional input for the user/site which is technically unnecessary.